search/util.go

```diff
@@ -67,3 +69,32 @@ func MergeFieldTermLocations(dest []FieldTermLocation, matches []*DocumentMatch)
 
 	return dest
 }
+
+const SearchIOStatsCallbackKey = "_search_io_stats_callback_key"
+
+type SearchIOStatsCallbackFunc func(uint64)
+
+// The callback signature is (message, queryType, cost) which allows
+// the caller to act on a particular query type and what its the associated
```

Comment:
Can we concise "query type and what its the associated cost of an operation." 
to
"query type and it's associated cost of operation"

Reply:
will do, thanks